apps/meteor/client/components/AutoCompleteDepartmentMultiple.tsx

```diff
@@ -1,34 +1,45 @@
 import { PaginatedMultiSelectFiltered } from '@rocket.chat/fuselage';
 import { useDebouncedValue } from '@rocket.chat/fuselage-hooks';
 import { useTranslation } from '@rocket.chat/ui-contexts';
+import type { FormEventHandler } from 'react';
 import React, { memo, useMemo, useState } from 'react';
 
 import { useRecordList } from '../hooks/lists/useRecordList';
 import { AsyncStatePhase } from '../hooks/useAsyncState';
 import { useDepartmentsList } from './Omnichannel/hooks/useDepartmentsList';
 
-const AutoCompleteDepartmentMultiple = (props) => {
-	const { value, onlyMyDepartments = false, showArchived = false, onChange = () => {} } = props;
+type AutoCompleteDepartmentMultipleProps = {
+	value?: string;
+	onChange: FormEventHandler<HTMLElement>;
+	onlyMyDepartments?: boolean;
+	showArchived?: boolean;
+};
 
+const AutoCompleteDepartmentMultiple = ({
+	value,
+	onlyMyDepartments = false,
+	showArchived = false,
+	onChange = () => undefined,
+}: AutoCompleteDepartmentMultipleProps) => {
 	const t = useTranslation();
 	const [departmentsFilter, setDepartmentsFilter] = useState('');
 
 	const debouncedDepartmentsFilter = useDebouncedValue(departmentsFilter, 500);
 
 	const { itemsList: departmentsList, loadMoreItems: loadMoreDepartments } = useDepartmentsList(
 		useMemo(
-			() => ({ filter: debouncedDepartmentsFilter, onlyMyDepartments, ...(showArchived && { showArchived: 'true' }) }),
+			() => ({ filter: debouncedDepartmentsFilter, onlyMyDepartments, ...(showArchived && { showArchived: true }) }),
 			[debouncedDepartmentsFilter, onlyMyDepartments, showArchived],
 		),
 	);
 
 	const { phase: departmentsPhase, items: departmentsItems, itemCount: departmentsTotal } = useRecordList(departmentsList);
 
 	const sortedByName = departmentsItems.sort((a, b) => {
-		if (a.name > b.name) {
+		if (a.label > b.label) {
 			return 1;
 		}
-		if (a.name < b.name) {
+		if (a.label < b.label) {
```

Comment:
Can't we skip sorting the data on UI? I see that we're already passing the sort param to the API so you should have the data already sorted at this point

Reply:
Indeed! That only shows how old is this hook, pushing changes